src/collective/taxonomy/behavior.py

```diff
@@ -70,9 +70,11 @@ def deactivateSearchable(self):
     def removeIndex(self):
         context = getSite()
         sm = context.getSiteManager()
-        sm.unregisterAdapter(TaxonomyIndexer, (IDexterityContent, IZCatalog),
-                             IIndexer, name=self.field_name)
-
+        old = sm._adapter_registrations.get(
+            ((IDexterityContent, IZCatalog), IIndexer, self.field_name), None)
```

Comment:
Can you avoid accessing a private api? Don't we have sm.queryAdapter here to check if there is a registered adapter?

Reply:
`sm.queryAdapter((IDexterityContent, IZCatalog), IIndexer, name=self.field_name)` maybe? Does it work?

Reply:
@vincentfretin ``queryAdapter`` does not work; actually, more correct would be ``queryMultiAdapter``, but that didn't work either, because those APIs expect objects, not interfaces as their first arguments.

With ``componentregistry.xml`` persistent adapters are also saved as items into ``portal._components`` so that the actually registered objects can be discovered easily for unregistration.

Reply:
@vincentfretin Oh, just looked sm.unregisterAdapter again, and it's enough to pass None as the existing factory to let it unregister any factory it finds. Updating pull soon.